hedera-node/src/main/java/com/hedera/services/txns/contract/process/EvmTxProcessor.java

```diff
@@ -154,17 +166,19 @@ protected TransactionProcessingResult execute(Account sender, Optional<Account>
 							.depth(0)
 							.completer(__ -> {
 							})
+							.isStatic(isStatic)
 							.miningBeneficiary(coinbase)
 							.blockHashLookup(h -> null);
 			final MessageFrame initialFrame = buildInitialFrame(commonInitialFrame,
-					receiver.map(acct -> acct.getId().asEvmAddress()).orElse(Address.ZERO), payload);
+					updater,
+					receiver, payload);
 			messageFrameStack.addFirst(initialFrame);
 
 			while (!messageFrameStack.isEmpty()) {
 				process(messageFrameStack.peekFirst(), new HederaTracer());
 			}
 
-			if (initialFrame.getState() == MessageFrame.State.COMPLETED_SUCCESS) {
+			if (initialFrame.getState() == MessageFrame.State.COMPLETED_SUCCESS && !isStatic) {
 				stackedUpdater.commit();
 			}
 
```

Comment:
Some of the variable computations are not required if `isStatic=true`. We could refactor the logic to the following one in order to skip unnecessary computation during static calls 
```Java
final Gas gasUsedByTransaction =
					Gas.of(gasLimit).minus(initialFrame.getRemainingGas());
			if (!isStatic) {
				/* Return leftover gas */
				final Gas selfDestructRefund =
						gasCalculator.getSelfDestructRefundAmount().times(initialFrame.getSelfDestructs().size());
				final Gas refundGas = initialFrame.getGasRefund().plus(selfDestructRefund);
				final Gas refunded = refunded(gasLimit, initialFrame.getRemainingGas(), refundGas);
				final Wei refundedWei = refunded.priceFor(Wei.of(gasPrice));
				mutableSender.incrementBalance(refundedWei);

				/* Send TX fees to coinbase */
				final var mutableCoinbase = updater.getOrCreate(coinbase).getMutable();
				final Gas coinbaseFee = Gas.of(gasLimit).minus(refunded);
				mutableCoinbase.incrementBalance(coinbaseFee.priceFor(Wei.of(gasPrice)));
				initialFrame.getSelfDestructs().forEach(updater::deleteAccount);

				/* Commit top level Updater */
				updater.commit();
			}
```